product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.1-registering-an-application/src/test/java/org/wso2/am/scenario/tests/register/application/ApplicationCreationNegativeTestCases.java

```diff
@@ -0,0 +1,183 @@
+//        }
+////        validate application wasn't created
+//        assertTrue(addApplicationJsonObject.getBoolean(ERROR),
+//                ERROR_DUPLICATE_APPLICATION_CREATION + APPLICATION_DUPLICATE_NAME);
+//        assertEquals(addApplicationJsonObject.get(MESSAGE).toString().trim(),
+//                ERROR_DUPLICATE_APPLICATION_EXIST + APPLICATION_DUPLICATE_NAME,
+//                ERROR_DUPLICATE_APPLICATION_CREATION + APPLICATION_DUPLICATE_NAME);
+    }
+
+//    todo uncomment once jappery api validation is fixed
+//    @Test(description = "4.1.1.6")
+//    public void testApplicationNameLongerThan70CharactersDataProvider() throws Exception {
+////        create application
+//        HttpResponse addApplicationResponse = apiStore
+//                .addApplication(URLEncoder.encode(APPLICATION_NAME_LONGER_THAN_70_CHARS, UTF_8),
+//                        URLEncoder.encode(APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED, UTF_8)
+//                        , "", URLEncoder.encode("", UTF_8));
+//        JSONObject addApplicationJsonObject = new JSONObject(addApplicationResponse.getData());
+////        if application added due to test failure add it to application list so that it could be removed later
+//        if (!addApplicationJsonObject.getBoolean(ERROR)
+//                && addApplicationJsonObject.get(STATUS).equals(STATUS_APPROVED)) {
+//            applicationsList.add(APPLICATION_NAME_LONGER_THAN_70_CHARS);
+//        }
+////        validate application wasn't created
+//        assertTrue(addApplicationJsonObject.getBoolean(ERROR),
+//                ERROR_APPLICATION_NAME_LONGER_THAN_70_CHARACTERS + APPLICATION_NAME_LONGER_THAN_70_CHARS);
+//    }
+
+    @Test(description = "4.1.1.7", dataProvider = "InvalidMandatoryApplicationValuesDataProvider",
+            dataProviderClass = ScenarioDataProvider.class)
+    public void testApplicationCreationWithInvalidMandatoryValues(String applicationName, String tier,
+                                                                  String description) throws Exception {
+//        create application
+        HttpResponse addApplicationResponse = apiStore
+                .addApplication(URLEncoder.encode(applicationName, UTF_8), URLEncoder.encode(tier, UTF_8)
+                        , "", URLEncoder.encode(description, UTF_8));
+        JSONObject addApplicationJsonObject = new JSONObject(addApplicationResponse.getData());
+//        if application added due to test failure add it to application list so that it could be removed later
+        if (!addApplicationJsonObject.getBoolean(ERROR)
+                && addApplicationJsonObject.get(STATUS).equals(STATUS_APPROVED)) {
+            applicationsList.add(applicationName);
+        }
+//        validate application wasn't created
+        assertTrue(addApplicationJsonObject.getBoolean(ERROR),
+                ERROR_IN_APP_CREATION_WITH_INVALID_INPUT + applicationName);
+    }
+
+    @AfterMethod(alwaysRun = true)
```

Comment:
Why is it AfterMethod here?

Reply:
In ApplicationCreationTestCases the methods testGenerateProductionKeysForApplication and testGenerateSandboxKeysForApplication depends on method testCreateApplicationForKeyGeneration. Therefore AfterMethod can't be used in that test case. Since there is no such dependency among test cases here i have used AfterMethod. We can change it to AfterClass if that is a better alternative.

Reply:
Okay. Let's keep it as it is.